lib/ash/filter/filter.ex

```diff
@@ -2708,46 +2708,43 @@ defmodule Ash.Filter do
        when is_atom(field) or is_binary(field) do
     cond do
       rel = relationship(context, field) ->
-        with relation_type when relation_type != :many_to_many <- rel.type,
-             true <- !Map.get(rel, :no_attributes),
-             [pk] when pk == rel.destination_attribute <-
-               Ash.Resource.Info.primary_key(rel.destination),
-             attr <- attribute(%{public?: true, resource: rel.source}, rel.source_attribute),
+        with rel_type when rel_type != :many_to_many <- rel.type,
+             dest <- rel.destination,
+             dest_attr <- rel.destination_attribute,
+             [pk] when pk == dest_attr <- Ash.Resource.Info.primary_key(dest),
+             attr <- attribute(%{public?: true, resource: dest}, pk),
              %Ash.Resource.Attribute{} = attr,
-             {:ok, result} <-
-               add_expression_part({attr.name, nested_statement}, context, expression) do
-          {:ok, result}
+             true <- is_list(nested_statement) or is_map(nested_statement) do
+          {:ok,
+           Enum.reduce(nested_statement, true, fn {key, val}, acc ->
+             {:ok, expr_part} =
+               if is_nil(aggregate(%{public?: true, resource: dest}, key)) and
```

Comment:
I think we actually want two branches here:

1. if the source_attribute is private, then we want to use the dest attribute.
2. otherwise, use this logic. But for now this should be fine 👍